File: AlgorithmStudy-vscode/Python/2204/9342.py
import re
import sys

input = sys.stdin.readline

testCase = int(input())
regex = re.compile(r"^[A-F]?A+F+C+[A-F]?$")

for i in range(testCase):
    text = input()

    if regex.match(text) != None:
        print("Infected!")
    else:
        print("Good")

File: Old Files/AlgorithmStudy.NET/Problems/2108/9342.cs
using System;
using System.Text.RegularExpressions;

namespace AlgorithmStudy.Problems._2108
{
    class _9342
    {
        static void Main()
        {
            Regex regex = new(@"^[A-F]?A+F+C+[A-F]?$");

            int count = int.Parse(Console.ReadLine());

            for (int i = 0; i < count; i++)
            {
                string input = Console.ReadLine();

                if (regex.IsMatch(input))
                    Console.WriteLine("Infected!");
                else
                    Console.WriteLine("Good");
            }
        }
    }
}
